Mu2eUtilities/inc/ReSeedByEventID.hh

```diff
@@ -0,0 +1,32 @@
+#ifndef Mu2eUtilities_ReSeedByEventID_hh
+#define Mu2eUtilities_ReSeedByEventID_hh
+//
+// Reseed a random engine using the event id and a user supplied salt.
+//
+
+#include "Offline/SeedService/inc/SeedService.hh"
+
+#include "canvas/Persistency/Provenance/EventID.h"
+#include "art/Framework/Services/Optional/RandomNumberGenerator.h"
+#include "CLHEP/Random/MixMaxRng.h"
+
+namespace mu2e {
+
+  class ReSeedByEventID{
+
+  public:
+    ReSeedByEventID( CLHEP::HepRandomEngine& engine, SeedService::seed_t salt);
+
+    void reseed ( art::EventID const& id ) const;
+
+  private:
+
+    CLHEP::MixMaxRng *  _engine; // the engine to be reseeded
+    SeedService::seed_t _salt;   // user supplied salt.
```

Comment:
Identifiers that start with an underscore  and are in the external scope are reserved  in gcc
https://www.gnu.org/software/libc/manual/html_node/Reserved-Names.html
Of course class member names are not global, but to a reader of a function that uses _salt it may not be obvious whether this is a member or a reserved global.     (And  underscore followed by a capital letter are reserved by C++ standard.)  I would advocate to just avoid names that start with an underscore in our new code;  use salt_ instead.

Reply:
The formatting got messed up:  I suggested to put the underscore at the end of salt instead of at the beginning